fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -361,6 +361,10 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
             components.core.contileTopSitesUpdater.startPeriodicWork()
         }
 
+        if (settings().enableUnifiedSearchSettingsUI) {
+            components.useCases.searchUseCases.restoreHiddenSearchEngines.invoke()
```

Comment:
This restore use case will run every time a user starts Fenix, right? Why do we need this to run all the time? From what I can see, this is a one-time change that we need to apply to correct users whose search engines were disabled.